src/redis_memory.rs

```diff
@@ -0,0 +1,242 @@
+use std::{
+    fmt::{self, Debug, Display},
+    hash::Hash,
+    marker::PhantomData,
+    sync::{Arc, Mutex},
+};
+
+use redis::{Commands, ToRedisArgs};
+
+use crate::MemoryADT;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct RedisMemoryError {
+    pub details: String,
+}
+
+impl std::error::Error for RedisMemoryError {}
+
+impl From<redis::RedisError> for RedisMemoryError {
+    fn from(err: redis::RedisError) -> Self {
+        RedisMemoryError {
+            details: err.to_string(),
+        }
+    }
+}
+
+impl Display for RedisMemoryError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Redis Memory Error: {}", self.details)
+    }
+}
+
+#[derive(Clone)]
+pub struct RedisMemory<Address, const WORD_LENGTH: usize>
+where
+    Address: Hash + Eq,
+{
+    connection: Arc<Mutex<redis::Connection>>,
+    _marker: PhantomData<Address>,
+}
+
+impl<Address: Hash + Eq, const WORD_LENGTH: usize> Debug for RedisMemory<Address, WORD_LENGTH> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("RedisMemory")
+            .field("connection", &"<redis::Connection>") // We don't want to debug the actual connection
+            .field("_marker", &self._marker)
+            .finish()
+    }
+}
+
+impl<Address: Hash + Eq, const WORD_LENGTH: usize> Default for RedisMemory<Address, WORD_LENGTH> {
+    fn default() -> Self {
+        Self {
+            connection: match redis::Client::open("redis://127.0.0.1:9999/") {
+                Ok(client) => match client.get_connection() {
+                    Ok(con) => Arc::new(Mutex::new(con)),
+                    Err(e) => {
+                        panic!("Failed to connect to Redis: {}", e);
+                    }
+                },
+                Err(e) => panic!("Error creating redis client: {:?}", e),
+            },
+            _marker: PhantomData,
+        }
+    }
+}
+
+/**
+ * Flushes the Redis database.
+ * WARNING: This is irreversible, do not run in production.
+ */
+#[cfg(test)]
+impl<Address: Hash + Eq + Debug, const WORD_LENGTH: usize> RedisMemory<Address, WORD_LENGTH> {
+    pub fn flush_db(&self) -> Result<(), redis::RedisError> {
+        let safe_connection = &mut *self.connection.lock().expect("Poisoned lock.");
+        redis::cmd("FLUSHDB").exec(safe_connection)?;
+        Ok(())
+    }
+}
+
+/**
+ * The RedisMemory implementation of the MemoryADT trait.
+ * All operations are - and should be - atomic.
+ */
```

Comment:
Use `///` to be consistent with the style of the repo.
use `--` instead of `-` for emphasis. Use back-quotes around variable/type/trait names

Otherwise, this doc carries no information.